draft-ietf-jsonpath-base.md

```diff
@@ -1592,6 +1663,54 @@ Change controller:
 Provisional registration? (standards tree only):
 : no
 
+## Function Extensions {#iana-fnex}
+
+This specification defines a new "Function Extensions sub-registry" in
+a new "JSONPath Parameters registry", with the policy "expert review"
+({{Section 4.5 of -ianacons}}).
+
+The experts are instructed to be frugal in the allocation of very short
+function extension names, keeping them in reserve for applications that
+are likely to enjoy wide use and can make good use of their shortness.
+The expert is also instructed to direct the registrant to provide a
+specification ({{Section 4.6 of -ianacons}}), but can make exceptions,
+for instance when a specification is not available at the time of
+registration but is likely forthcoming.
+If the expert becomes aware of function extensions that are deployed and
+in use, they may also initiate a registration on their own if
+they deem such a registration can avert potential future collisions.
+{: #de-instructions}
+
+Each entry in the registry must include:
+
+{:vspace}
+Function Name:
+: a lower case ASCII {{-ascii}} string that starts with a letter and can
+  contain letters, digits and underscore characters afterwards
+  (`[a-z][_a-z0-9]*`).
```

Comment:
So we're only accepting English-character function names?  Just clarifying.

Reply:
Well, ASCII, yes.
This is an interesting question in a more general way, as so far JSONPath is mostly indifferent to languages.
But these function extensions are intended for an international audience, so it is probably best to do a lowest common denominator.

Reply:
Obviously, I stole the ABNF from dot-member-name, which allows %x80-10FFFF -- this is a recipe for confusion, with RTL characters, format effectors, and all that.  I'm having a hard time making a suggestion as to what we should allow there -- `\p{L}|\p{N}` maybe, in parallel to allowing ALPHA + DIGIT?  But then we also allow "_", and there may be equivalent separators in some scripts.  So this is hard to get useful over a wide domain, while reducing the potential for confusion.  function names of course do not depend on whatever language was used in the input data item, so we can be more frugal in what we allow.

Reply:
No further discussion -- I'm assuming we stick with `[a-z][_a-z0-9]*` (no change required)